pkg/utils/addresses.go

```diff
@@ -230,20 +248,32 @@ func addressesDefaultInternal(preferIPv6 bool, af AddressFilter, getAddrs addres
 			log.Debugf("Address %s is on interface %s with default route", address, link.Attrs().Name)
 			// We should only have one default route per interface
 			addrs = append(addrs, FoundAddress{
-				Address:   address.IP,
-				Priority:  routeMap[linkIndex][0].Priority,
-				LinkIndex: linkIndex,
+				Address:         address.IP,
+				Priority:        routeMap[linkIndex][0].Priority,
+				LinkIndex:       linkIndex,
+				HasDefaultRoute: address.IPNet.Contains(routeMap[linkIndex][0].Gw),
 			})
 		}
 	}
 
-	// Sort addresses into a stable order, based on default route priority and link
+	// Sort addresses into a stable order, based on:
+	// a) default route priority
+	// b) link
+	// c) IP address family
+	// d) The default route matching the IP CIDR
+	// e) The address being public
 	// index. Otherwise the order of the addresses we return may change if an address
```

Comment:
Nit: The "index." here is a leftover from the now-removed comment.

Reply:
Changed + renamed one of the variables